simplemonitor/Monitors/remote_ssh.py

```diff
@@ -0,0 +1,135 @@
+import paramiko
+import re
+from typing import Tuple, cast
+
+
+class Operator(Enum):
+    EQUALS = "equals"
+    NOT_EQUALS = "not_equals"
+    GREATER_THAN = "greater_than"
+    LESS_THAN = "less_than"
+
+
+class OperatorType(Enum):
+    STRING = "str"
+    INTEGER = "int"
+
+
+@register
+class MonitorRemoteSSH(Monitor):
+
+    monitor_type = "remote_ssh"
+
+    def __init__(self, name: str, config_options: dict) -> None:
+        super().__init__(name, config_options)
+        # description
+        self.description = cast(str, self.get_config_option("description", required=False))  # maybe define default here instead of a try: ?
+        self.success_message = cast(str, self.get_config_option("success_message", required=False, default="it worked"))
+        # ssh configuration
+        self.command = cast(str, self.get_config_option("command", required=True))
+        self.ssh_private_key_path = cast(str, self.get_config_option("ssh_private_key_path", required=True))
+        self.ssh_username = cast(str, self.get_config_option("ssh_username", required=True))
+        self.target_hostname = cast(str, self.get_config_option("target_hostname", required=True))
+        self.target_port = cast(int, self.get_config_option("target_port", required=False, default="22"))
+        # operator logic
+        self.operator = cast(
+            str,
+            self.get_config_option(
+                "operator",
+                required=True,
+                allowed_values=[Operator.EQUALS.value, Operator.NOT_EQUALS.value, Operator.LESS_THAN.value, Operator.GREATER_THAN.value],
+            ),
+        )
+        self.regex = re.compile(cast(str, self.get_config_option("regex", required=True)))
+        # values to compare, cast to the expected type
+        self.result_type = cast(
+            str,
+            self.get_config_option("result_type", required=True, allowed_values=[OperatorType.INTEGER.value, OperatorType.STRING.value]),
+        )
+        match self.result_type:
+            case OperatorType.INTEGER.value:
+                self.target_value = cast(int, self.get_config_option("target_value", required=True))
+            case OperatorType.STRING.value:
+                self.target_value = cast(str, self.get_config_option("target_value", required=True))
+
+    def run_test(self) -> bool:
+        # run remote command
+        client = paramiko.SSHClient()
+        client.set_missing_host_key_policy(paramiko.AutoAddPolicy)
+        try:
+            client.connect(
+                self.target_hostname,
+                username=self.ssh_username,
+                key_filename=self.ssh_private_key_path,
+                port=self.target_port,
+            )
+        except TimeoutError:
+            return self.record_fail(f"connection to {self.target_hostname} timed out")
+        except ConnectionRefusedError:
+            return self.record_fail(f"connection to {self.target_hostname} actively refused")
+        except Exception as e:
```

Comment:
Perhaps Paramiko's `SSHException` instead of `Exception` here?

Reply:
fixed within https://github.com/jamesoff/simplemonitor/commit/074399d1f935f527d47dc0f45f1c212787917f91